source/MaterialXGenGlsl/GlslShaderGenerator.h

```diff
@@ -37,6 +37,9 @@ class MX_GENGLSL_API GlslShaderGenerator : public HwShaderGenerator
     /// Return the version string for the GLSL version this generator is for
     virtual const string& getVersion() const { return VERSION; }
 
+    /// Emit function definitions for light sampling code
+    virtual void emitLightSamplerFunctionDefinitions(const ShaderGraph& graph, GenContext& context, ShaderStage& stage) const;
```

Comment:
Should we expose this in Python and Javascript ?

Reply:
I don't think that is necessary. The reason we add this is for custom generators to be able to override the generation of light functions. Calling a generator from Python/Javascript make sense, but for developing a new custom generator I think one needs to use C++. We could even make this method protected as it's only called from the generator itself.

Reply:
One additional (nitpick) comment is that maybe we could remove "Sampler" from the metod name, and just name it emitLightFunctionDefinitions()

Reply:
+1 on making protected and method rename as it's more than just the sampler code which is being emitted.